### PR TITLE
feat(data-fetchers): migrate VCF data-fetcher to TS

### PR DESCRIPTION
**Pull request by manzt**
Follow up to #739

### CODE REVIEW

File: src/data-fetcher/utils.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+    return info;
+}
+
+export type ExtendedChromInfo = HiGlass.ChromInfo & {
+    absToChr(absPos: number): ReturnType<typeof absToChr> | null;
+    chrToAbs(chr: [name: string, pos: number]): number | null;
+};
+
+export async function fetchChromInfo(url: string): Promise<ExtendedChromInfo> {
```
Comment (manzt) [inline src/data-fetcher/utils.ts:71]: We should be able to reuse this module for the BAM worker and others.

File: src/data-fetcher/vcf/gosling-vcf-data.ts [hunk 2]
```diff
@@ -3,41 +3,34 @@
...
     private uid: string;
-    private fetchTimeout: any;
-    private toFetch: Set<any>;
+    private assembly: Assembly;
+    private toFetch: Set<string>;
+    private fetchTimeout?: ReturnType<typeof setTimeout>;
     private track: any;
-    private prevRequestTime: number;
-    private tbiIndexed: any;
-    private tbiVCFParser: any;
```
Comment (manzt) [inline src/data-fetcher/vcf/gosling-vcf-data.ts:26]: I only kept properties which are assigned in the constructor.

File: src/missing-types.d.ts [hunk 3]
```diff
@@ -312,10 +315,11 @@ declare module '@higlass/utils' {
     import type { ScaleContinuousNumeric } from 'd3-scale';
     import type { TilesetInfo } from '@higlass/services';
 
-    type ChromInfo<Name extends string> = {
+    export type ChromInfo<Name extends string = string> = {
```
Comment (sehilyi) [inline src/missing-types.d.ts]: I think we can reuse this in our assembly definitions as well:

https://github.com/gosling-lang/gosling.js/blob/db1b657f3918272759236c1317c0ab00fc4acfb2/src/core/utils/assembly.ts#L12

**Review by sehilyi — APPROVED** [review]
Looks good to me!

File: src/missing-types.d.ts (hunk 3 above)
Comment (manzt) [reply src/missing-types.d.ts]: Ok great!

**Comment by manzt** [comment]
Hold on... I think I know how to make these diffs nicer..

**Pull request merged**